src/js/outreach/views/verify_views.js

```diff
@@ -0,0 +1,124 @@
+import hbs from 'handlebars-inline-precompile';
+import { View } from 'marionette';
+
+import 'scss/modules/buttons.scss';
+import 'scss/modules/forms.scss';
+
+import './verify.scss';
+
+const RequestCodeView = View.extend({
+  ui: {
+    submit: '.js-submit',
+  },
+  triggers: {
+    'click @ui.submit': 'click:submit',
+  },
+  template: hbs`
+    <div class="verify__icon verify__icon--blue">{{fat "user-lock"}}</div>
+    <h2 class="verify__heading-text">Request a verification code to view this health resource.</h2>
+    <p class="verify__info-text">We’ll send a text message with a verification code to the phone number 123-456-7890.</p>
+    <button class="verify__submit button--green w-100 js-submit">Send Verification Code</button>
+  `,
+  onClickSubmit() {
+    this.ui.submit.prop('disabled', true);
+  },
+});
+
+const VerifyCodeView = View.extend({
+  ui: {
+    submit: '.js-submit',
+    resend: '.js-resend',
+    input: '.js-input',
+  },
+  triggers: {
+    'click @ui.submit': 'click:submit',
+    'click @ui.resend': 'click:resend',
+  },
+  events: {
+    'input @ui.input': 'watchInput',
+    'keydown @ui.input': 'watchKeydown',
+  },
+  template: hbs`
+    <div class="verify__icon verify__icon--blue">{{fat "user-lock"}}</div>
+    <h2 class="verify__heading-text">Enter your verification code.</h2>
+    <p class="verify__info-text">We sent a text message with a verification code to the phone number 123-456-7890.</p>
+    <div class="verify__code-fields">
+      <input class="input-primary verify__code-input js-input" />
+      <input class="input-primary verify__code-input js-input" />
+      <input class="input-primary verify__code-input js-input" />
+      <input class="input-primary verify__code-input js-input" />
+    </div>
+    <button class="verify__submit button--green w-100 js-submit" disabled>Send Verification Code</button>
+    <div class="verify__heading-text u-text-link js-resend">Send a new code</div>
+  `,
+  watchInput(event) {
+    const inputElements = this.$('.js-input');
+    const index = inputElements.index(event.target);
+
+    const [first, ...rest] = event.target.value;
```

Comment:
this is to get the first character correct?   
I'm on the fence on the destructuring.. makes it look like an array.. maybe string functions would be clearer?
```js
const value = String(event.target.value);
const first = event.target.value = value.charAt(0);
const rest = value.substr(1);
```

Reply:
Updated this to not do the destructuring.